monkestation/code/modules/surgery/organs/internal/heart.dm

```diff
@@ -131,6 +139,50 @@
 		return
 	to_chat(H, span_warning("...but there is not enough of you to go around! You must attain more mass to heal!"))
 
+/**
+ * Allows oozelings to selectively retract a limb.
+ */
+/datum/action/innate/retract_limb
+	name = "Retract Limb"
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "retract_limb"
+	button_icon = SLIME_ACTIONS_ICON_FILE
+	background_icon_state = "bg_alien"
+
+/datum/action/innate/retract_limb/IsAvailable(feedback)
+	. = ..()
+	if(!.)
+		return
+	if(!isoozeling(owner))
+		return
+	var/mob/living/carbon/human/user = owner
+	var/list/limbs = list(user.get_bodypart(BODY_ZONE_R_ARM), user.get_bodypart(BODY_ZONE_L_ARM), user.get_bodypart(BODY_ZONE_R_LEG), user.get_bodypart(BODY_ZONE_L_LEG))
+	if(!length(limbs))
+		return FALSE // What are you gonna eat if there's nothing left?
+	return TRUE
+
+/datum/action/innate/retract_limb/Activate()
+	. = ..()
+	if(!isoozeling(owner))
+		return
+	var/mob/living/carbon/human/user = owner
+	var/list/possible_limbs = list(BODY_ZONE_HEAD, BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)
+	var/list/retractable_limbs = list()
+	for(var/zone as anything in possible_limbs)
+		var/obj/item/bodypart/limb = user.get_bodypart(zone)
+		if(!isnull(limb))
+			retractable_limbs[limb] = image(getFlatIcon(limb))
```

Comment:
```suggestion
			retractable_limbs[limb] = new /mutable_appearance(limb)
```

you do _not_ need getflaticon here

Reply:
in fact does `[limb`] = limb` not work? i swear i remember radial menus could just use atoms directly as icons

Reply:
Limbs don't have an icon state.

Reply:
appearance cloning would also copy overlays